tests/real/enterprise.rs

```diff
@@ -321,3 +321,43 @@ fn ent_unique_job_until_start() {
 
     handle.join().expect("should join successfully");
 }
+
+#[test]
+fn ent_unique_job_bypass_unique_lock() {
+    use faktory::error;
+    use serde_json::Value;
+
+    skip_if_not_enterprise!();
+
+    let url = learn_faktory_url();
+    let mut producer = Producer::connect(Some(&url)).unwrap();
+
+    let job1 = Job::builder("order")
+        .queue("ent_unique_job_bypass_unique_lock")
+        .unique_for(60)
+        .build();
+
+    // Now the following job is _technically_ a 'duplicate', BUT if the `unique_for` value is not set,
+    // the uniqueness lock will be bypassed on the server. This special case is mentioned in the docs:
+    // https://github.com/contribsys/faktory/wiki/Ent-Unique-Jobs#bypassing-uniqueness
+    let job2 = Job::builder("order") // same jobtype and args (args are just not set)
+        .queue("ent_unique_job_bypass_unique_lock") // same queue
+        .build(); // NB: `unique_for` not set
+
+    producer.enqueue(job1).unwrap();
+    producer.enqueue(job2).unwrap(); // bypassing the lock!
+
+    // This _is_ a 'duplicate'.
+    let job3 = Job::builder("order")
+        .queue("ent_unique_job_bypass_unique_lock")
+        .unique_for(60) // NB
+        .build();
+
+    let res = producer.enqueue(job3).unwrap_err(); // NOT bypassing the lock!
+
+    if let error::Error::Protocol(error::Protocol::UniqueConstraintViolation { msg }) = res {
+        assert_eq!(msg, "Job not unique");
+    } else {
+        panic!("Expected protocol error.")
+    }
```

Comment:
Should we also check that both `job1` and `job2` _were_ enqueued?

Reply:
Will be addressed in [PR](https://github.com/jonhoo/faktory-rs/pull/46)